src/System.Net.Http/src/System/Net/Http/Headers/HttpHeaderValueCollection.cs

```diff
@@ -34,7 +34,7 @@ namespace System.Net.Http.Headers
     //   - Property TransferEncodingChunked: is set to "true".
     public sealed class HttpHeaderValueCollection<T> : ICollection<T> where T : class
     {
-        private string _headerName;
+        private HeaderDescriptor _descriptor;
```

Comment:
They are created on demand for collection-valued headers.

I believe that using the HeaderDescriptor infrastructure, I can get rid of the _validator field -- looking at that now.  Future PR though.